qcodes/instrument_drivers/american_magnetics/AMI430.py

```diff
@@ -340,15 +340,19 @@ def set_field(self,
 
         # Otherwise, wait until no longer ramping
         self.log.debug(f'Starting blocking ramp of {self.name} to {value}')
-        while self.ramping_state() == 'ramping':
-            self._sleep(0.3)
-        self._sleep(2.0)
-        state = self.ramping_state()
+        exit_state = self.wait_while_ramping()
         self.log.debug(f'Finished blocking ramp')
         # If we are now holding, it was successful
-        if state != 'holding':
+        if exit_state != 'holding':
             msg = '_set_field({}) failed with state: {}'
-            raise AMI430Exception(msg.format(value, state))
+            raise AMI430Exception(msg.format(value, exit_state))
+
+    def wait_while_ramping(self, check_interval=0.05) -> str:
```

Comment:
i'd suggest to expose this interval as an attribute (or even as a parameter) so that anyone can change that as needed without needing to change the code. @nikhartman would you agree?

Reply:
That makes sense, this is likely dependent on what interface is being used (TCP/Serial/...)